### PR TITLE
[Feature] add ranged directional light

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: cocos/3d/lights/ranged-directional-light-component.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright (c) 2022-2022 Xiamen Yaji Software Co., Ltd.
```
Comment (minggo) [inline cocos/3d/lights/ranged-directional-light-component.ts]: error date

File: cocos/render-scene/scene/ranged-directional-light.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    private _scale: Vec3 = new Vec3(1, 1, 1);
+    private _right: Vec3 = new Vec3(1, 0, 0);
+    private _illuminanceHDR: number = Ambient.SUN_ILLUM;
+    private _illuminanceLDR = 1.0;
+
+    /**
+     * @en The direction vector of the light
+     * @zh 光源的方向
+     */
+    get direction (): Vec3 {
```
Comment (minggo) [inline cocos/render-scene/scene/ranged-directional-light.ts]: I think should add readonly. As we don't want game codes modify it outside which can not work on native too.

**Review by minggo — CHANGES_REQUESTED** [review]

File: cocos/render-scene/scene/ranged-directional-light.ts (hunk 2 above)
Comment (troublemaker52025) [reply cocos/render-scene/scene/ranged-directional-light.ts]: sure

**Review by minggo — APPROVED** [review]

**Review by jk20012001 — APPROVED** [review]

**Pull request merged**